interp/cfg.go

```diff
@@ -69,7 +69,12 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 				// type of array like value is not yet known. This could be fixed in ast structure
 				// by setting array/map node as 1st child of ForRangeStmt instead of 3rd child of
 				// RangeStmt. The following workaround is less elegant but ok.
-				if t := sc.rangeChanType(n.anc); t != nil {
+				var t *itype
```

Comment:
It would be better to test that the channel type is not a `chanSendT` early and break with error, rather than modify `rangeChanType`.

Also do not forget about the possibility of a binary channel (`typ.cat == valueT`) send-only: `typ.TypeOf().Kind() == reflect.Chan && typ.TypeOf().ChanDir == reflect.SendDir`. And also the fact that the type could be aliased (`typ.cat == aliasT`).

For all those reasons, it's better to have an helper `isSendChan` method on type, and  test early as proposed above. You can find other similar helper in type.go, i.e. `isChan` or `isInterface`, etc.

Reply:
Alright, will do, thanks !